config/configuration_singleton.rb

```diff
@@ -0,0 +1,96 @@
+#
+# to avoid the traditional singleton approach or using class methods, both of
+# which make it difficult to write tests against
+#
+# instead, ConfigurationSingleton is the definition of the configuration
+# then the singleton instance used is a new class called "Configuration" which
+# we set in config/boot i.e.
+#
+# Configuration = ConfigurationSingleton.new
+#
+# This is functionally equivalent to taking every instance method on
+# ConfigurationSingleton and defining it as a class method on Configuration.
+#
+class ConfigurationSingleton
+  # The environment
+  # @return [String] "development", "test", or "production"
+  def rails_env
+    ENV['RAILS_ENV'] || ENV['RACK_ENV'] || "development"
+  end
+
+  # The app's root directory
+  # @return [Pathname] path to app root
+  def app_root
+    Pathname.new(File.expand_path("../../",  __FILE__))
+  end
+
+  # The app's configuration root directory
+  # @return [Pathname] path to configuration root
+  def config_root
+    Pathname.new(ENV["OOD_APP_CONFIG_ROOT"] || "/etc/ood/config/apps/osc-reporting")
+  end
+
+  FALSE_VALUES = [nil, false, '', 0, '0', 'f', 'F', 'false', 'FALSE', 'off', 'OFF', 'no', 'NO'].freeze
+
+  # Bool coersion pulled from ActiveRecord::Type::Boolean#cast_value
+  #
+  # @return [Boolean] false for falsy value, true for everything else
+  def to_bool(value)
+    !FALSE_VALUES.include?(value)
+  end
+
+  def load_external_config?
+    to_bool(ENV['OOD_LOAD_EXTERNAL_CONFIG'] || (rails_env == 'production'))
+  end
+
+  def dotenv_local_files
+    [
+      app_root.join(".env.#{rails_env}.local"),
+      (app_root.join(".env.local") unless rails_env == "test"),
+    ].compact
+  end
+
+  def dotenv_files
+    [
+      (config_root.join("env") if load_external_config?),
+      app_root.join(".env.#{rails_env}"),
+      app_root.join(".env")
+    ].compact
+  end
+
+  # reverse list and suffix every path with '.overload'
+  def overload_files(files)
+    files.reverse.map {|p| p.sub(/$/, '.overload')}
+  end
+
+  # Load the dotenv local files first, then the /etc dotenv files and
+  # the .env and .env.production or .env.development files.
+  #
+  # Doing this in two separate loads means OOD_APP_CONFIG_ROOT can be specified in
+  # the .env.local file, which will specify where to look for the /etc dotenv
+  # files. The default for OOD_APP_CONFIG_ROOT is /etc/ood/config/apps/myjobs and
+  # both .env and .env.production will be searched for there.
+  def load_dotenv_files
+    # .env.local first, so it can override OOD_APP_CONFIG_ROOT
+    Dotenv.load(*dotenv_local_files)
+
+    # load the rest of the dotenv files
+    Dotenv.load(*dotenv_files)
+
+    # load overloads
+    Dotenv.overload(*(overload_files(dotenv_files)))
+    Dotenv.overload(*(overload_files(dotenv_local_files)))
+  end
+
+  # Minutes to wait between cluster_status reloads
+  def minutes_between_cluster_status_reload
+    ENV['MINUTES_BETWEEN_CLUSTER_STATUS_RELOAD'].to_i || 1
```

Comment:
One thing is we try to use the `OOD_` prefix for environment variables.  Plus once you get into 4 word territory, you gotta wonder how can we say this in less?

```suggestion
  def cluster_reload_delay
    ENV['OOD_CLUSTER_RELOAD_DELAY'].to_i || 1
```